weblogic-azure-aks/src/main/arm/scripts/createVMAndBuildImage.sh

```diff
@@ -0,0 +1,121 @@
+
+# read <azureACRPassword> and <ocrSSOPSW> from stdin
+function read_sensitive_parameters_from_stdin() {
+    read azureACRPassword ocrSSOPSW
+}
+
+function cleanup_vm() {
+    #Remove VM resources
+    az extension add --name resource-graph
+    # query vm id
+    vmId=$(az graph query -q "Resources \
+| where type =~ 'microsoft.compute/virtualmachines' \
+| where name=~ '${vmName}' \
+| where resourceGroup  =~ '${currentResourceGroup}' \
+| project vmid = id" -o tsv)
+
+    # query nic id
+    nicId=$(az graph query -q "Resources \
+| where type =~ 'microsoft.compute/virtualmachines' \
+| where name=~ '${vmName}' \
+| where resourceGroup  =~ '${currentResourceGroup}' \
+| extend nics=array_length(properties.networkProfile.networkInterfaces) \
+| mv-expand nic=properties.networkProfile.networkInterfaces \
+| where nics == 1 or nic.properties.primary =~ 'true' or isempty(nic) \
+| project nicId = tostring(nic.id)" -o tsv)
+
+    # query ip id
+    ipId=$(az graph query -q "Resources \
+| where type =~ 'microsoft.network/networkinterfaces' \
+| where id=~ '${nicId}' \
+| extend ipConfigsCount=array_length(properties.ipConfigurations) \
+| mv-expand ipconfig=properties.ipConfigurations \
+| where ipConfigsCount == 1 or ipconfig.properties.primary =~ 'true' \
+| project  publicIpId = tostring(ipconfig.properties.publicIPAddress.id)" -o tsv)
+
+    # query os disk id
+    osDiskId=$(az graph query -q "Resources \
+| where type =~ 'microsoft.compute/virtualmachines' \
+| where name=~ '${vmName}' \
+| where resourceGroup  =~ '${currentResourceGroup}' \
+| project osDiskId = tostring(properties.storageProfile.osDisk.managedDisk.id)" -o tsv)
+
+    # query vnet id
+    vnetId=$(az graph query -q "Resources \
+| where type =~ 'Microsoft.Network/virtualNetworks' \
+| where name=~ '${vmName}VNET' \
+| where resourceGroup  =~ '${currentResourceGroup}' \
+| project vNetId = id" -o tsv)
+
+    # query nsg id
+    nsgId=$(az graph query -q "Resources \
+| where type =~ 'Microsoft.Network/networkSecurityGroups' \
+| where name=~ '${vmName}NSG' \
+| where resourceGroup  =~ '${currentResourceGroup}' \
+| project nsgId = id" -o tsv)
+
+    # Delete VM NIC IP VNET NSG resoruces
+    vmResourceIdS=$(echo ${vmId} ${nicId} ${ipId} ${osDiskId} ${vnetId} ${nsgId})
+    echo ${vmResourceIdS}
+    az resource delete --verbose --ids ${vmResourceIdS}
+}
+
+# Build docker image
+#  * Create Ubuntu machine VM-UBUNTU
+#  * Running vm extension to run buildWLSDockerImage.sh, the script will:
+#    * build a docker image with domain model, applications based on specified WebLogic Standard image
+#    * push the image to ACR
+function build_docker_image() {
+    # Create vm to build docker image
+    vmName="VM-UBUNTU-WLS-AKS-$(date +%s)"
+
+    # MICROSOFT_INTERNAL
+    # Specify tag 'SkipASMAzSecPack' to skip policy 'linuxazuresecuritypackautodeployiaas_1.6'
+    # Specify tag 'SkipNRMS*' to skip Microsoft internal NRMS policy, which cause vm-redeployed issue
```

Comment:
cause -> causes

Reply:
Fixed in https://github.com/galiacheng/weblogic-azure/commit/e599233edb163bdfa597ea2551aee0771f364913